From 1a32e3e74ad855a0b85650b08a4b0947edbf383b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 14 Sep 2019 14:33:09 +0200
Subject: [PATCH 106/152] volume: Clarify some code

We have more idiomatic ways to check whether any element fullfills
some condition than breaking out of a loop.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/731
---
 js/ui/status/volume.js | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/js/ui/status/volume.js b/js/ui/status/volume.js
index 21968107d..db8e5a1f1 100644
--- a/js/ui/status/volume.js
+++ b/js/ui/status/volume.js
@@ -259,18 +259,17 @@ var InputStreamSlider = class extends StreamSlider {
     _maybeShowInput() {
         // only show input widgets if any application is recording audio
         let showInput = false;
-        let recordingApps = this._control.get_source_outputs();
-        if (this._stream && recordingApps) {
-            for (let i = 0; i < recordingApps.length; i++) {
-                let outputStream = recordingApps[i];
-                let id = outputStream.get_application_id();
-                // but skip gnome-volume-control and pavucontrol
-                // (that appear as recording because they show the input level)
-                if (!id || (id != 'org.gnome.VolumeControl' && id != 'org.PulseAudio.pavucontrol')) {
-                    showInput = true;
-                    break;
-                }
-            }
+        if (this._stream) {
+            // skip gnome-volume-control and pavucontrol which appear
+            // as recording because they show the input level
+            let skippedApps = [
+                'org.gnome.VolumeControl',
+                'org.PulseAudio.pavucontrol'
+            ];
+
+            showInput = this._control.get_source_outputs().some(output => {
+                return !skippedApps.includes(output.get_application_id());
+            });
         }
 
         this._showInput = showInput;
-- 
2.23.0

